File: sdc-sdks/sdc-sdk-javascript/src/api/ContaApi.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient', '../model/Conta', '../model/ResponseEntity'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/Conta'), require('../model/ResponseEntity'));
  } else {
    // Browser globals (root is window)
    if (!root.SDC) {
      root.SDC = {};
    }
    root.SDC.ContaApi = factory(root.SDC.ApiClient, root.SDC.Conta, root.SDC.ResponseEntity);
  }
}(this, function(ApiClient, Conta, ResponseEntity) {
  'use strict';

  /**
   * Conta service.
   * @module api/ContaApi
   * @version 1.0.0
   */

  /**
   * Constructs a new ContaApi. 
   * @alias module:api/ContaApi
   * @class
   * @param {module:ApiClient} apiClient Optional API client implementation to use, default to {@link module:ApiClient#instance}
   * if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the createUsingPOST1 operation.
     * @callback module:api/ContaApi~createUsingPOST1Callback
     * @param {String} error Error message, if any.
     * @param {module:model/Conta} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Cria uma conta
     * Cria uma conta
     * @param {module:model/Conta} conta conta
     * @param {module:api/ContaApi~createUsingPOST1Callback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/Conta}
     */
    this.createUsingPOST1 = function(conta, callback) {
      var postBody = conta;

      // verify the required parameter 'conta' is set
      if (conta == undefined || conta == null) {
        throw "Missing the required parameter 'conta' when calling createUsingPOST1";
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['access_token'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = Conta;

      return this.apiClient.callApi(
        '/v1/contas', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the deleteUsingDELETE operation.
     * @callback module:api/ContaApi~deleteUsingDELETECallback
     * @param {String} error Error message, if any.
     * @param {module:model/ResponseEntity} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Deleta uma conta
     * Deleta um cart\u00EF\u00BF\u00BDo
     * @param {Integer} id id
     * @param {module:api/ContaApi~deleteUsingDELETECallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/ResponseEntity}
     */
    this.deleteUsingDELETE = function(id, callback) {
      var postBody = null;

      // verify the required parameter 'id' is set
      if (id == undefined || id == null) {
        throw "Missing the required parameter 'id' when calling deleteUsingDELETE";
      }


      var pathParams = {
        'id': id
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['access_token'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = ResponseEntity;

      return this.apiClient.callApi(
        '/v1/contas/{id}', 'DELETE',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the getAllUsingGET1 operation.
     * @callback module:api/ContaApi~getAllUsingGET1Callback
     * @param {String} error Error message, if any.
     * @param {Array.<module:model/Conta>} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Retorna todas as contas
     * Retorna todas as contas
     * @param {module:api/ContaApi~getAllUsingGET1Callback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {Array.<module:model/Conta>}
     */
    this.getAllUsingGET1 = function(callback) {
      var postBody = null;


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['access_token'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = [Conta];

      return this.apiClient.callApi(
        '/v1/contas', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the getOneUsingGET1 operation.
     * @callback module:api/ContaApi~getOneUsingGET1Callback
     * @param {String} error Error message, if any.
     * @param {module:model/Conta} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Retorna uma conta
     * Retorna uma conta
     * @param {Integer} id id
     * @param {module:api/ContaApi~getOneUsingGET1Callback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/Conta}
     */
    this.getOneUsingGET1 = function(id, callback) {
      var postBody = null;

      // verify the required parameter 'id' is set
      if (id == undefined || id == null) {
        throw "Missing the required parameter 'id' when calling getOneUsingGET1";
      }


      var pathParams = {
        'id': id
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['access_token'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = Conta;

      return this.apiClient.callApi(
        '/v1/contas/{id}', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the updateUsingPUT1 operation.
     * @callback module:api/ContaApi~updateUsingPUT1Callback
     * @param {String} error Error message, if any.
     * @param {module:model/Conta} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Atualiza uma conta
     * Atualiza uma conta
     * @param {module:model/Conta} conta conta
     * @param {module:api/ContaApi~updateUsingPUT1Callback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/Conta}
     */
    this.updateUsingPUT1 = function(conta, callback) {
      var postBody = conta;

      // verify the required parameter 'conta' is set
      if (conta == undefined || conta == null) {
        throw "Missing the required parameter 'conta' when calling updateUsingPUT1";
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['access_token'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = Conta;

      return this.apiClient.callApi(
        '/v1/contas', 'PUT',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));

File: sdc-sdks/sdc-sdk-android/src/main/java/br/com/conductor/sdc/api/v1/ContaApi.java
package br.com.conductor.sdc.api.v1;

import br.com.conductor.sdc.api.v1.invoker.ApiException;
import br.com.conductor.sdc.api.v1.invoker.ApiInvoker;
import br.com.conductor.sdc.api.v1.invoker.Pair;

import br.com.conductor.sdc.api.v1.model.*;

import java.util.*;

import br.com.conductor.sdc.api.v1.model.Conta;
import br.com.conductor.sdc.api.v1.model.Response;

import org.apache.http.HttpEntity;
import org.apache.http.entity.mime.MultipartEntityBuilder;

import java.util.Map;
import java.util.HashMap;
import java.io.File;

public class ContaApi {
  String basePath = "https://localhost/";
  ApiInvoker apiInvoker = ApiInvoker.getInstance();

  public void addHeader(String key, String value) {
    getInvoker().addDefaultHeader(key, value);
  }

  public ApiInvoker getInvoker() {
    return apiInvoker;
  }

  public void setBasePath(String basePath) {
    this.basePath = basePath;
  }

  public String getBasePath() {
    return basePath;
  }

  
  /**
   * Cria uma conta
   * Cria uma conta
   * @param conta conta
   * @return Conta
   */
  public Conta  createUsingPOST1 (Conta conta) throws ApiException {
    Object postBody = conta;
    
    // verify the required parameter 'conta' is set
    if (conta == null) {
       throw new ApiException(400, "Missing the required parameter 'conta' when calling createUsingPOST1");
    }
    

    // create path and map variables
    String path = "/v1/contas".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    String[] contentTypes = {
      "application/json"
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder builder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = builder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      
    }

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "POST", queryParams, postBody, headerParams, formParams, contentType);
      if(response != null){
        return (Conta) ApiInvoker.deserialize(response, "", Conta.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Deleta uma conta
   * Deleta um cart\u00C3\u00A3o
   * @param id id
   * @return Response
   */
  public Response  deleteUsingDELETE (Long id) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'id' is set
    if (id == null) {
       throw new ApiException(400, "Missing the required parameter 'id' when calling deleteUsingDELETE");
    }
    

    // create path and map variables
    String path = "/v1/contas/{id}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "id" + "\\}", apiInvoker.escapeString(id.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    String[] contentTypes = {
      "application/json"
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder builder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = builder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      
    }

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "DELETE", queryParams, postBody, headerParams, formParams, contentType);
      if(response != null){
        return (Response) ApiInvoker.deserialize(response, "", Response.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Retorna todas as contas
   * Retorna todas as contas
   * @return List<Conta>
   */
  public List<Conta>  getAllUsingGET1 () throws ApiException {
    Object postBody = null;
    

    // create path and map variables
    String path = "/v1/contas".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    String[] contentTypes = {
      "application/json"
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder builder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = builder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      
    }

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
      if(response != null){
        return (List<Conta>) ApiInvoker.deserialize(response, "array", Conta.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Retorna uma conta
   * Retorna uma conta
   * @param id id
   * @return Conta
   */
  public Conta  getOneUsingGET1 (Long id) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'id' is set
    if (id == null) {
       throw new ApiException(400, "Missing the required parameter 'id' when calling getOneUsingGET1");
    }
    

    // create path and map variables
    String path = "/v1/contas/{id}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "id" + "\\}", apiInvoker.escapeString(id.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    String[] contentTypes = {
      "application/json"
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder builder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = builder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      
    }

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
      if(response != null){
        return (Conta) ApiInvoker.deserialize(response, "", Conta.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Atualiza uma conta
   * Atualiza uma conta
   * @param conta conta
   * @return Conta
   */
  public Conta  updateUsingPUT1 (Conta conta) throws ApiException {
    Object postBody = conta;
    
    // verify the required parameter 'conta' is set
    if (conta == null) {
       throw new ApiException(400, "Missing the required parameter 'conta' when calling updateUsingPUT1");
    }
    

    // create path and map variables
    String path = "/v1/contas".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();

    

    

    String[] contentTypes = {
      "application/json"
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder builder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = builder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      
    }

    try {
      String response = apiInvoker.invokeAPI(basePath, path, "PUT", queryParams, postBody, headerParams, formParams, contentType);
      if(response != null){
        return (Conta) ApiInvoker.deserialize(response, "", Conta.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
}
